src/core/sort.cc

```diff
@@ -1433,19 +1433,21 @@ RiGb group(const std::vector<Column>& columns,
   const Column& col0 = columns[0];
 
   size_t nrows = col0.nrows();
-  #if DT_DEBUG
-    for (const Column& col : columns) xassert(col.nrows() == nrows);
-  #endif
+  size_t n_const_cols = 0;
+  for (const Column& col : columns) {
+    xassert(col.nrows() == nrows);
+    n_const_cols += col.is_constant();
+  }
 
   // For a 0-row Frame we return a rowindex of size 0, and the
   // Groupby containing zero groups.
   if (nrows == 0) {
     result.second = Groupby::zero_groups();
     return result;
   }
-  if (nrows == 1) {
-    Buffer buf = Buffer::mem(sizeof(int32_t));
-    buf.set_element<int32_t>(0, 0);
+  if (nrows == 1 || n_const_cols == n) {
```

Comment:
So this will only skip sorting only when all the columns are constant, right? If we have two columns to be sorted, and only one of them is constant — we will still sort both columns?

Reply:
yes. It can be improved, as you have noted, but will take more work to account for sortflags etc.